### DISCUSSION TITLE
y dimension (latitude) not found

### DISCUSSION THREAD

**Discussion by shuai-zhou** [corteva/rioxarray#468] [Q&A] [answered]
I was processing NDVI data (one of the files can be downloaded [here](https://github.com/shuai-zhou/DataRepo/blob/7a3cc4cb48dbacac563043f75907c96d567e60ed/data/AVHRR-Land_v005_AVH13C1_NOAA-19_20100101_c20170406091314.nc?raw=true)). The data looks like the following:

![f1](https://user-images.githubusercontent.com/22311821/153335899-3e451509-bd4f-459f-8d01-6e8c5d31fa96.png)

However, when I was writing the crs using `ds.rio.write_crs('epsg:4326', inplace=True)`. I got the error `MissingSpatialDimensionError: y dimension (latitude) not found. Data variable: lon_bnds`. I googled and tried to fix this but did not make it.

**Comment by snowman2** [comment, marked as answer]
Did you use `decode_coords="all"` when opening the file? It looks like it already has a `crs` coordinate that may already have the necessary information.

https://corteva.github.io/rioxarray/stable/getting_started/getting_started.html

**Reply by snowman2** [reply]
Also, what version of rioxarray are you using?

**Reply by shuai-zhou** [reply]
Hi, I was using rioxarray version 0.7.1. While `decode_coords="all"` solved the problem above, another error occurred when I was clipping the gridded data using `clipped = ds.rio.clip(poly4326.geometry.apply(mapping), poly4326.crs)`. The error is `ValueError: Could not convert object to NumPy datetime`. But the `time` is already in `datetime64[ns]` format...

**Reply by snowman2** [reply]
See #459. Might be worth a separate discussion.

**Reply by shuai-zhou** [reply]
Thanks for the help. Yes, it turns out that there are missing values in the `ds`; the missingness is not on the `time` variable but on `TIMEOFDAY` and `QA`. Dropping the two variables solve the problem.
